### PR TITLE
Improve "I follow the ... link in the email" step

### LINKED ISSUE #110: "I follow the ... link in the email" should only follow <a> links in HTML e-mails
The above step from Spreewald scans for any URL-like string and tries to visit it. This is too much for HTML e-mails where it hits `xmlns` URLs or similar.

Instead, the step should do something like this:

```ruby
When /^I follow the (first|second|third)? ?link in the e?mail$/ do |index_in_words|
  mail = @mail || ActionMailer::Base.deliveries.last
  index = { nil => 0, 'first' => 0, 'second' => 1, 'third' => 2 }[index_in_words]
  url_pattern = %r{(?:http|https)://[^/]+([^"'\s\\]*)}

  paths = if mail.html_part
    dom = Nokogiri::HTML(mail.html_part.body.to_s)
    (dom / 'a[href]').map { |a| a['href'].match(url_pattern)[1] }
  else
    mail_body = MailFinder.email_text_body(mail).to_s
    mail_body.scan(url_pattern)[0]
  end

  visit paths[index]
end
```

### PR DESCRIPTION
**Pull request by foobear**
Resolves #110

### CODE REVIEW

**Comment by foobear** [comment]
Test matrix passing.

Maybe we'd want to add Rails 5 or 6 and a current Ruby at some point. I'll open a separate issue for that.

File: lib/spreewald/email_steps.rb [hunk 1]
```diff
@@ -69,9 +69,16 @@
...
-  mail_body = MailFinder.email_text_body(mail).to_s
-  only_path = mail_body.scan(url_pattern)[index][0]
-  visit only_path
+
+  paths = if mail.html_part
+    dom = Nokogiri::HTML(mail.html_part.body.to_s)
+    (dom / 'a[href]').map { |a| a['href'].match(url_pattern)[1] }
+  else
+    mail_body = MailFinder.email_text_body(mail).to_s
+    mail_body.scan(url_pattern)[0]
```
Comment (makmic) [inline lib/spreewald/email_steps.rb]: It's a bit hard to parse/understand why you are using `.match(pattern)[1]` for HTML parts and `.scan(pattern)[0]` for text parts.

Comment (makmic) [reply lib/spreewald/email_steps.rb]: Ok after reading the docs I get it, `match` includes the entire match as the array's first entry.
Please unify the use in this case :slightly_smiling_face:

Comment (foobear) [reply lib/spreewald/email_steps.rb]: Valid point, will adjust :+1:

Comment (foobear) [reply lib/spreewald/email_steps.rb]: The implementation is actually incorrect since I changed it to `scan(...)[0]`. Will fix.

Comment (foobear) [reply lib/spreewald/email_steps.rb]: Actually can't use the same method in both cases since they handle different approaches (one loops over HTML hyperlinks, the other one parses the entire text body).

However, I've cleaned it up a bit and added a test for the text body part (and fixed it).
